document/core/appendix/algorithm.rst

```diff
@@ -354,15 +366,51 @@ Other instructions are checked in a similar manner.
          push_vals(label_types(ctrls[n]))
          push_val(Ref(rt.heap, false))
 
-       case (call_ref)
-         let rt = pop_ref()
-         if (rt.heap =/= Bot)
-           error_if(not is_def(rt.heap))
-           let ft = funcs[rt.heap.idx]
-           pop_vals(ft.params)
-           push_vals(ft.results)
+       case (br_on_cast n rt1 rt2)
+         validate_ref_type(rt1)
```

Comment:
is it ok that the new helper functions here, `validate_ref_type`, and `is_func/struct` etc, are never defined above?